staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere_util.go

```diff
@@ -608,6 +609,119 @@ func (vs *VSphere) checkDiskAttached(ctx context.Context, nodes []k8stypes.NodeN
 	return nodesToRetry, nil
 }
 
+// BuildMissingVolumeNodeMap builds a map of volumes and nodes which are not known to attach detach controller
+func (vs *VSphere) BuildMissingVolumeNodeMap(ctx context.Context) {
+	nodeDetails := vs.nodeManager.GetNodeNames()
+	// Segregate nodes according to VC-DC
+	dcNodes := make(map[string][]k8stypes.NodeName)
+
+	for _, nodeName := range nodeDetails {
+		// if given node is not in node volume map
+		if !vs.vsphereVolumeMap.CheckForNode(nodeName) {
+			nodeInfo, err := vs.nodeManager.GetNodeInfo(nodeName)
```

Comment:
I'm trying to understand where/how we would learn the name of an unknown node. It doesn't look to me like you are listing all VMs in a datacenter (that would be a lot!)

Reply:
I will add some additional details and I updated PR description. Basically this is a follow up of https://github.com/kubernetes/kubernetes/pull/96224, because I found a bug where if a node has no pods with volumes scheduled on it, we don't run `VerifyVolumesAreAttached` on that node and as a result any dangling volume left on such node is never detached.

So strictly speaking this PR isn't detaching volumes from "unknown" nodes but more like nodes which aren't in attach-detach controller's cache (because there are no pods with volume on those nodes).

Reply:
Also, I am only listing VMs that are still part of k8s cluster.